### ISSUE TITLE
How to connect to raspberry

### ISSUE THREAD

**Issue by nikybiasion** [voroshkov/Solo-DIY-RF-Laptimer#9]
I see that i need to use easyracelaptimer 0.6 and raspberry pi 3. I suppose i've to connect pin 5 from arduino to the GPIO pin as described in easyracelaptimer vtx_sensor docs.
But, arduino works at 5v and raspberry at 3,3v. I need a voltage divider? Arduino and Raspberry need to share the same GND? I'm not an expert electronic :)

**Comment by voroshkov** [comment]
I did it like this (used some N-type MOSFET which opens at less than 5V).
Also you may use 3.3V to 5V logic level converter, but in this case Arduino code should be changed to generate a pulse of inverted polarity.

![erlt_wiring](https://cloud.githubusercontent.com/assets/8790856/22662767/2e467f8a-ecbb-11e6-8094-6f829083873b.png)

**Comment by voroshkov** [comment]
Also here's the visual reference to Raspberry Pin numbers. 
ERLT uses pins:
- gpio25: 'VTX_1' - pin 37
- gpio24: 'VTX_2' - pin 35
- gpio23: 'VTX_3' - pin 33
- gpio22: 'VTX_4' - pin 31
- gpio21: 'VTX_5' - pin 29
- gpio3:  'VTX_6' - pin 15
- gpio2:  'VTX_7' - pin 13
- gpio0:  'VTX_8' - pin 11

![wiringpi](https://cloud.githubusercontent.com/assets/8790856/22663154/9cf50bc6-ecbc-11e6-8ad4-f4db55df5412.png)

**Comment by R91g** [comment]
> I see that i need to use easyracelaptimer 0.6 and raspberry pi 3. I suppose i've to connect pin 5 from arduino to the GPIO pin as described in easyracelaptimer vtx_sensor docs.
But, arduino works at 5v and raspberry at 3,3v. I need a voltage divider? Arduino and Raspberry need to share the same GND? I'm not an expert electronic :)

I don't understand.
Raspberry works at 5v, you don't need any voltage dividir, you can power Raspberry and Arduino from the same 5v.

**Comment by voroshkov** [comment]
Raspberry is powered from 5V, but GPIO levels are 3.3V. 
The question was about connecting Solo laptimer device to Raspberry GPIO to use it as a VTX sensor for Easy Race Lap Timer software.

**Comment by R91g** [comment]
Yes, me too I want connect Solo Laptimer Device to Raspberry Pi with Easy Race Timer.
I I believed that only need to connect Arduino Pin 5 to GPIO 37, and also power the raspberry Pi and Arduino from a baterry at 5v

**Comment by R91g** [comment]
Another question.

Can we use a Arduino Pro Mini 3.3v 8Mhz?
Are 8Mhz enough for run this code properly?

**Comment by voroshkov** [comment]
> Can we use a Arduino Pro Mini 3.3v 8Mhz?
Are 8Mhz enough for run this code properly?

I think there's nothing in code which depends on board frequency, but I didn't try. Also timing accuracy will suffer a bit. 

>With this arduino we can connect pin 5 to Raspberry Pi directly. In this case we can't power RX5808 from arduino, but we can power it from Raspberry Pi GPIO 5v.

But in this case I'm not sure RX5808 will correctly interpret SPI signals from Arduino.

**Comment by shamshir1972** [comment]
voroshkov, may i know N type mosfet parts number that u use. I've no idea what parts number to buy

**Comment by voroshkov** [comment]
I got mine from old laptop motherboard. You can use any "logical" mosfet, e.g. BSS138 or powerful IRF630.

**Comment by FranzAr** [comment]
So I think, this is clear so far. But I still have to use the app to set band/channel, Threshold and minimum lap time. Is this correct or can/could it be done via EasyRacelapTimer software?

**Comment by voroshkov** [comment]
Yes, you have to maintain 2 connections (wired to Raspberry, and bluetooth to configure). 
I didn't investigate the possibility to use ERLT for configuration, and don't plan it so far.
If anyone's ready to spend time for investigation, I'll eagerly help with explanation of communication protocol.

**Comment by asficas** [comment]
Voroshkov You have done an excellent job! Thank you! 
I built a twin solo chorus and we raced one on one in the car park and it worked really well! 
I am now looking at easyracelaptimer pocket edition serial protocol and will try to connect the two solos to raspi3 ERLT. I will let you know if I make any progress.
Thanks again from Athens Greece.

**Comment by voroshkov** [comment]
Cool! Thanks!

**Comment by shamshir1972** [comment]
voroshkov, i've already setup voltage divider using IRF630 as u suggested but i cannot get any rssi value at erlt racebox webpage. Is it i've to configure anything ? i'm using 0.6 image file, pls help me

**Comment by voroshkov** [comment]
You won't get any RSSI value in ERLT. You need to setup device via Bluetooth and then start race in ERLT. ERLT will just capture time when RSSI is higher than threshold (when drone passes the laptimer).

**Comment by shamshir1972** [comment]
ok i got it working, i've already set threshold via bluetooth b4, but in pilot name under token id i've to set as 'VTX_1' instead of "VTX_SENSOR_1" to make erlt triggered the lap

**Comment by voroshkov** [comment]
can you specify what is bluetooth b4?
